state.go

```diff
@@ -0,0 +1,110 @@
+package frankenphp
+
+import (
+	"slices"
+	"strconv"
+	"sync"
+)
+
+type stateID int
```

Comment:
Nit: should be large enough:

```suggestion
type stateID uint8
```